build_debian.sh

```diff
@@ -249,6 +249,10 @@ sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y in
     locales                 \
     cgroup-tools
 
+## Note: temporary workaround to allow snmp container to import smbus for arista
+sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install      \
```

Comment:
Should we install python3-smbus to the  snmp docker itself?  
e,g, change this file: 
https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-snmp-sv2/Dockerfile.j2

Reply:
As you can see this Dockerfile extends docker-config-engine which extends docker-base which is jessie based. In jessie there is no such thing as python3-smbus. Stretch will have it once all the docker images will be upgraded to this version.

Also there will be another blocker since we build our own version of python3.6 for snmp. Stretch currently uses python3.5. It likely means that installing python3-smbus will only make it available for python3.5 and not python3.6. 

I tried to confine the hack to the snmp container by manually installing the stretch python3-smbus in jessie but that generated a lot of errors so I abandoned this idea. We will be tracking the evolution of all these component and remove these mitigations once we can find a sound solution.

Reply:
Could you please  use 'pip3.6 install' it from pypi? Then you don't worry about python3.5/3.6, nor about jessie/stretch.

---
In reply to: [274686749](https://github.com/Azure/sonic-buildimage/pull/2772#discussion_r274686749) [](ancestors = 274686749)

Reply:
I should have left a bigger comment for this change.
The smbus module is a cpython module and therefore require at least build-essentials and python-dev to compile as part of `pip3.6 install`.
I felt it was overkill to install the build toolchain install the module and remove the toolchain in the Dockerfile.

Reply:
Actually we have similar logic in dockers/docker-snmp-sv2/Dockerfile.j2 already:
1. install gcc
2. pip install hiredis/pyyaml
3. remove gcc
Could you please try this `overkill' method? I prefer it because of binary compatibility issue between Jessie container and Stretch host.

---
In reply to: [274727333](https://github.com/Azure/sonic-buildimage/pull/2772#discussion_r274727333) [](ancestors = 274727333)